src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -0,0 +1,671 @@
+                " cells must be greater than the provided minimum and less than 1.");
+        this.maxCopyNumber = ParamUtils.isPositive(maxCopyNumber, "Maximum copy number must be positive");
+
+        /* auxiliary member functions */
+        this.minHetAlleleFraction = (1 - this.maxAbnormalFraction) / (this.maxCopyNumber * this.maxAbnormalFraction +
+                2 * (1 - this.maxAbnormalFraction));
+        this.breakpointHetAlleleFraction = (1 - this.minAbnormalFraction) / (this.maxCopyNumber * this.minAbnormalFraction +
+                2 * (1 - this.minAbnormalFraction));
+
+        /* initialize the integration quadrature and calculate the allele fraction prior on the abscissas */
+        initializeIntegrationQuadrature(ParamUtils.isPositive(quadratureOrder - MIN_QUADRATURE_ORDER,
+                "Quadrature order must be greater than " + MIN_QUADRATURE_ORDER) + MIN_QUADRATURE_ORDER);
+        initializeHetAlleleFractionPrior();
+        logger.info("Using the HETEROGENEOUS Het prior.");
+    }
+
+    /******************************
+     * core computational methods *
+     ******************************/
+
+    /**
+     * Initilizes the quadrature for calculating allele ratio integrals in getHetLogLikelihood
+     * @param numIntegPoints  number of points in the quadrature
+     */
+    private void initializeIntegrationQuadrature(final int numIntegPoints) {
+
+        /* get Gauss-Legendre quadrature factory of order @numIntegPoints */
+        GaussIntegratorFactory integratorFactory = new GaussIntegratorFactory();
+        GaussIntegrator gaussIntegrator = integratorFactory.legendre(numIntegPoints,
+                minHetAlleleFraction, 1.0 - minHetAlleleFraction);
+
+        /* abscissas */
+        gaussIntegrationAbscissas.clear();
+        gaussIntegrationAbscissas.addAll(IntStream.range(0, numIntegPoints).
+                mapToDouble(gaussIntegrator::getPoint).boxed().collect(Collectors.toList()));
+
+        /* weights */
+        gaussIntegrationWeights.clear();
+        gaussIntegrationWeights.addAll(IntStream.range(0, numIntegPoints).
+                mapToDouble(gaussIntegrator::getWeight).boxed().collect(Collectors.toList()));
+
+        /* log of weights */
+        gaussIntegrationLogWeights.clear();
+        gaussIntegrationLogWeights.addAll(gaussIntegrationWeights.stream().
+                mapToDouble(FastMath::log).boxed().collect(Collectors.toList()));
+    }
+
+    /**
+     * (advanced) calculate a simple prior (and log prior) for the allele fraction based on
+     * (1) minimum purity of the sample, and (2) maximum copy number. See CNV-methods.pdf for details.
+     */
+    private void initializeHetAlleleFractionPrior() {
+
+        alleleFractionPriors.clear();
+        alleleFractionPriors.addAll(gaussIntegrationAbscissas.stream()
+                .mapToDouble(alleleFraction -> {
```

Comment:
Please extract this Function into its own private method (e.g. ```calculateAlleleFractionPrior```) and then make reference to it ```.mapToDouble(this::calculateAlleleFractionPrior)```.

Will make the code more readable.